src/common/snippets/include/snippets/lowered/expression.hpp

```diff
@@ -90,13 +91,13 @@ class IOExpression : public Expression {
 public:
     enum class io_type {INPUT, OUTPUT, UNDEFINED};
     ExpressionPtr clone_with_new_inputs(const std::vector<PortConnectorPtr>& new_inputs,
-                                        const std::shared_ptr<Node>& new_node) const override;
+                                        const std::shared_ptr<Node>& new_node, bool deep_shape_clone = true) const override;
     int64_t get_index() const  { return m_index; }
     io_type get_type() const { return m_type; }
     // Result needs shapeInfer to copy shape from Parent's output to this expr input
     bool needShapeInfer() const override {return m_type == io_type::OUTPUT; }
 private:
-    IOExpression(const IOExpression& other) = default;
+    IOExpression(const IOExpression& other, bool deep_shape_clone = true);
```

Comment:
I agree that it's ugly interface. But I did so to make opportunity to clone LinearIR with the same `shape_ptr` inline. To avoid additional function calls after copy of LinearIR.
After your argument with `subtensor`, I totally agree that it's not scalable approach. Thanks a lot for the idea!